src/PowerShellEditorServices.Protocol/Server/LanguageServer.cs

```diff
@@ -1216,22 +1218,34 @@ private bool IsQueryMatch(string query, string symbolName)
                                 Arguments = JArray.FromObject(correction.Edits)
                             });
                     }
+                }
+            }
 
-                    if (string.Equals(diagnostic.Source, "PSScriptAnalyzer", StringComparison.OrdinalIgnoreCase))
-                    {
-                        codeActionCommands.Add(
-                            new CodeActionCommand
-                            {
-                                Title = $"Show documentation for \"{diagnostic.Code}\"",
-                                Command = "PowerShell.ShowCodeActionDocumentation",
-                                Arguments = JArray.FromObject(new[] { diagnostic.Code })
-                            });
-                    }
+            // Add "show documentation" commands last so they appear at the bottom of the client UI. 
+            // These commands do not require code fixes. Sometimes we get a batch of diagnostics
+            // to create commands for. No need to create multiple show doc commands for the same rule.
+            var ruleNamesProcessed = new HashSet<string>();
+            foreach (var diagnostic in codeActionParams.Context.Diagnostics)
+            {
+                if (string.IsNullOrEmpty(diagnostic.Code)) { continue; }
+
+                if (string.Equals(diagnostic.Source, "PSScriptAnalyzer", StringComparison.OrdinalIgnoreCase) &&
+                    !ruleNamesProcessed.Contains(diagnostic.Code))
+                {
+                    ruleNamesProcessed.Add(diagnostic.Code);
+
+                    codeActionCommands.Add(
+                        new CodeActionCommand
+                        {
+                            Title = $"Show documentation for \"{diagnostic.Code}\"",
+                            Command = "PowerShell.ShowCodeActionDocumentation",
+                            Arguments = JArray.FromObject(new[] { diagnostic.Code })
+                        });
                 }
             }
 
             await requestContext.SendResult(
-                codeActionCommands.ToArray());
+            codeActionCommands.ToArray());
```

Comment:
nit: re-indent or put on one line

Reply:
Yeah, short enough to be one line.  I'll fix that.  Not sure how that got un-indented.  Hmm...

Reply:
Doh! Missed this one.  I'll fix it tonight.